sunpy/net/fido_factory.py

```diff
@@ -91,10 +91,7 @@ def _handle_record_slice(self, client_resp, record_slice):
 
         # Make sure we always have an iterable, as most of the response objects
         # expect one.
-        if isinstance(record_slice, int):
-            resp = [client_resp[record_slice]]
-        else:
-            resp = client_resp[record_slice]
+        resp = list(client_resp[record_slice])
```

Comment:
I don't think this logic is quite the same as what it is removing. For example `list('hello')` isn't the same operation as `['hello']` given I think I have changed this block in #3770 I propose you revert this change.

Reply:
For single-element slicing, the previous implementation works but fails for multi-element slicing.
Eg: If we do slicing as `res[0, 0]`, then `resp` would be a list of `JSOCResponse`. But if we do slicing as `res[0, 0:2]`, then it gives a `Table`. In that case, we have to handle `JSOCResponse` differently. 
So I thought to maintain the uniformity, we should cast all of the slicings as a list.

Reply:
I think the issue is

> But if we do slicing as res[0, 0:2], then it gives a Table.

not how we are handling it here.

I am not 100% sure though, it's hard to think about this without really jumping into it.